bin/hubot.js

```diff
@@ -8,7 +8,8 @@ const OptParse = require('optparse')
 const Hubot = require('..')
 
 const switches = [
-  ['-a', '--adapter ADAPTER', 'The Adapter to use'],
+  ['-a', '--adapter ADAPTER', 'The Adapter to use, e.g. "shell" (to load the default hubot shell adapter)'],
+  ['-p', '--path PATH', 'Path to adapter directory, e.g. "./adapters" (Note: if this is set, the adapter option will be used as the file name to load from this folder)'],
```

Comment:
With this change, the `-a` behaves one way when `-p` is active, and another way when it doesn't. This feels like it exposes the user to complexity in order to save addition of an extra switch. I'd suggest to be explicit both internally and externally.

I recommend that each switch should have a single defined purpose, rather than switching behaviour based on context. (I'm trying to think of another CLI tool which demonstrates the behaviour here, and can't, which to me suggests this isn't the way to do it.)

I would be more comfortable with a separate switch for "full path to adapter", or to make `-p` accept a full path to the adapter and override `-a` (that feels less complex to me).

Reply:
I concur. I'll take another stab at it.

Reply:
I'd also add that `--path` is general enough that it wouldn't be super obvious

Perhaps we can re-use `--adapter` to accept a name of an adapter, or a directory? It'd be possible to check if the value given is a directory on the filesystem to distinguish the two.